### DISCUSSION TITLE
Unable to video stream to drone engage

### DISCUSSION THREAD

**Discussion by ievbayiro32** [DroneEngage/droneengage_communication#3] [Q&A]
Unable to video stream to drone engage through USB camera and Raspberry Pi 3. I've been attempting this for almost 2 months and have fully determined it is a software issue as I've streamed to 2 cameras already.

I have noticed in my efforts to install and untinstall drone engage onto my pi, the de_camera.so is different for me. I noticed it is de_camera64.so, and it tells me "can't execute binaries"

I also can guarantee it is NOT one of the follpowing common issues because I have either redone these steps several times, or have been able to stream using the pi's IP.

-enabling legacy camera support 
-camera functionality (have been able to stream to pi IP through USB and flex connnection cameras)

<img width="797" alt="Screenshot 2024-03-31 182900" src="https://github.com/DroneEngage/droneengage_communication/assets/165714210/5348c593-c0aa-4c91-aeda-70965dbc7f75">
<img width="561" alt="Screenshot 2024-03-31 182730" src="https://github.com/DroneEngage/droneengage_communication/assets/165714210/9542d066-09ec-48cf-8742-ef13e1825abb">

**Comment by HefnySco** [comment]
kindly review this thread [https://github.com/DroneEngage/droneengage_communication/discussions/2](https://github.com/DroneEngage/droneengage_communication/discussions/2)

**Reply by HefnySco** [reply]
please use **BullsEye**

**Reply by ievbayiro32** [reply]
I'm using bullseye software, here's also extra images of what I'm getting.
![3](https://github.com/DroneEngage/droneengage_communication/assets/165714210/f5f035f3-2ba2-4c65-a3f8-a7d969d25119)
![2](https://github.com/DroneEngage/droneengage_communication/assets/165714210/bd9e8262-37bd-4df1-994d-eda47327cce5)
![1](https://github.com/DroneEngage/droneengage_communication/assets/165714210/b7f148e6-f60b-4498-8a5f-2bdf8a6838ed)
![bullseye](https://github.com/DroneEngage/droneengage_communication/assets/165714210/087f0e07-374d-47e6-9721-d7cd0fb34d4d)

**Comment by HefnySco** [comment]
You can either use:

```
    // Camera Devices Configuration
      "camera_start_index"        :0, 
      "camera_end_index"          :10, 

```      
where the app will scan /dev/video0 till /dev/video10 and list all video devices that it finds regardless it is a camera or not.

      or use
      
      "camera_list": [
        {"name": "ONE", "device_num": 0} // name should be unique across all cameras.
        ]
      
   where you point to a specific device and give your camera a name.
   
   

In your example you  use "camera_list" but you comment the entry inside it that is why it is not working.
Also please comment 
```
//      "camera_start_index"        :0, 
//      "camera_end_index"          :10,  
```

Thanks

**Reply by ievbayiro32** [reply]
I tried that configuration with no success for weeks. I honestly think a video for camera setup is necessary because there could be specific discreptancies disrupting a system. If you see the screenshot attached you will see both configurations that I have tried, where both have failed
![Screenshot 2024-04-03 101828](https://github.com/DroneEngage/droneengage_communication/assets/165714210/d2dd3752-1065-4c98-b502-1ce6995056cf)
![Screenshot 2024-04-03 101759](https://github.com/DroneEngage/droneengage_communication/assets/165714210/a15bed53-bda4-4ae1-a7e5-ffaed0b3b06b)
![Screenshot 2024-04-03 101709](https://github.com/DroneEngage/droneengage_communication/assets/165714210/73b33bcf-6a22-4cdd-8438-b2c98afd4361)
![Screenshot 2024-04-03 101638](https://github.com/DroneEngage/droneengage_communication/assets/165714210/8f547fb5-41d3-40a8-92e4-c115e0fcd933)

**Comment by HefnySco** [comment]
please note that camera icon should be green not grey.

**Reply by ievbayiro32** [reply]
I know that, I don't know why it's still grey. I have my usb camera connected

**Reply by HefnySco** [reply]
if it is grey then the camera is not detected, or the camera module cannot detect de_communicator. but in your case I believe it is the first one.
From the WebClient in the Details tab you can see the running modules and their version.

**Reply by ievbayiro32** [reply]
![Capture](https://github.com/DroneEngage/droneengage_communication/assets/165714210/53613619-ca10-4e1d-b783-abb543d04b07)
I was able to get the web client to detect the camera(s) but nothing is appearing on screen. I do know the camera work because I used fswebcam to take the following images. For reference also, I'm using drone engage for a senior team project. We are making a autononmous boat from scratch.
![test](https://github.com/DroneEngage/droneengage_communication/assets/165714210/65bce144-f923-4d79-a2fb-3762f06e855f)

**Reply by HefnySco** [reply]
gery camera icon means camera module is not detected by communicator module.
grey camera screen means webrtc communication cannot be established.

**Comment by HefnySco** [comment]
Ok I did a fresh installation on RPI4.
I am using this exact version **2023-02-21-raspios-bullseye-arm64-lite.img(2023-02-21 04:14).xz**

```
$ uname -a
Linux raspberrypi 5.15.84-v8+ #1613 SMP PREEMPT Thu Jan 5 12:03:08 GMT 2023 aarch64 GNU/Linux

```
I connected an USB Camera 


```
$ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 003: ID 046d:0825 Logitech, Inc. Webcam C270
Bus 001 Device 002: ID 2109:3431 VIA Labs, Inc. Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

and I  downloaded this version of DroneEngage:
[https://cloud.ardupilot.org/downloads/RPI/latest/64bit/Drone_Engage_07Jan_2024.zip](https://cloud.ardupilot.org/downloads/RPI/latest/64bit/Drone_Engage_07Jan_2024.zip)
and this version of Camera module:
[https://cloud.ardupilot.org/downloads/RPI/latest/camera/Drone_Engage_Camera_07Jan_2024.zip](https://cloud.ardupilot.org/downloads/RPI/latest/camera/Drone_Engage_Camera_07Jan_2024.zip)

And everything worked from the first time.... I just run the installation and even used the default account.


![image](https://github.com/DroneEngage/droneengage_communication/assets/4184373/5c2c7d65-4dfa-4df7-b4bd-6b219a4f9de9)

`cat ./config.module.json`

![image](https://github.com/DroneEngage/droneengage_communication/assets/4184373/94cf05be-a5f2-472c-9555-60042e268280)

Please make sure that camera service is OFF before running the following line:

you can check status using:

sudo systemctl status de_camera.service

![image](https://github.com/DroneEngage/droneengage_communication/assets/4184373/9ab76fd1-b1fc-42eb-b59b-53d229fd7e7b)


and you need to stop it to run the file manually using command:

`sudo systemctl stop de_camera.service
`

To Start the camera module from any location use the following command:

`/home/pi/drone_engage/de_camera/de_camera64.so  -c /home/pi/drone_engage/de_camera/config.module.json
`



![image](https://github.com/DroneEngage/droneengage_communication/assets/4184373/7096b49b-1ac5-4827-8f43-5d4c6e5f09c5)

**Reply by ievbayiro32** [reply]
Good afternoon, I look at all of your new 
![Screenshot 2024-04-05 152624](https://github.com/DroneEngage/droneengage_communication/assets/165714210/2e25ea0d-958a-4bb4-9520-5b60ce667d8e)
![Screenshot 2024-04-05 152733](https://
![Screenshot 2024-04-05 152759](https://github.com/DroneEngage/droneengage_communication/assets/165714210/9f047bdd-f7f3-447a-b434-6c8b7ff93634)
github.com/DroneEngage/droneengage_communication/assets/165714210/4b8e07c9-e303-425f-a5a6-7a205995ced5)

![Screenshot 2024-04-05 152616](https://github.com/DroneEngage/droneengage_communication/assets/165714210/285ab33d-a6aa-478b-9e69-93ae08fddcd3)
advice and here is what I got on my end

**Reply by ievbayiro32** [reply]
Good afternoon, I looked at all of your new advice and here is what I got on my end

**Reply by HefnySco** [reply]
please check my uname:

$ uname -a
Linux raspberrypi 5.15.84-v8+ #1613 SMP PREEMPT Thu Jan 5 12:03:08 GMT 2023 aarch64 GNU/Linux

**Reply by HefnySco** [reply]
I am using this exact version 2023-02-21-raspios-bullseye-arm64-lite.img(2023-02-21 04:14).xz

**Reply by HefnySco** [reply]
Also could you try with a normal RPI cam, or just change this cam to one with lower resolution... just try

**Reply by ievbayiro32** [reply]
pi@raspberrypi:~ $ uname -a
Linux raspberrypi 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux

Also, I attached a RPi Camera, it also isn't streaming . I attached an image of both drone engage screens for both cameras, the USB one and RPi one. I also attached an image of the RPi Camera that shows its working b/c of the red record led on it but it isn't showing on drone engage.

![Screenshot 2024-04-06 131020](https://github.com/DroneEngage/droneengage_communication/assets/165714210/02906a79-a343-4e9b-be38-226ac31405fe)
![Screenshot 2024-04-06 131015](https://github.com/DroneEngage/droneengage_communication/assets/165714210/0923c2e2-dc91-4a11-bf6e-37f184d1e5e5)
![Screenshot 2024-04-06 125550](https://github.com/DroneEngage/droneengage_communication/assets/165714210/d9b8d76b-7e8e-4b3b-adeb-c9188b6893f7)
![IMG_6457](https://github.com/DroneEngage/droneengage_communication/assets/165714210/5ec1bf38-c5f9-4e75-8885-0eed649f8e8c)

**Reply by ievbayiro32** [reply]
Also I may think this is something to note. 
![Screenshot 2024-04-06 131848](https://github.com/DroneEngage/droneengage_communication/assets/165714210/9036ccb6-b2d3-4b3c-89ac-8735f4614104)

**Comment by HefnySco** [comment]
So after trying this I can say video is working fine.
The only missing issue is display capture photo from DroneEngage which will be fixed very soon.

**Reply by DroneEngage** [reply]
Display capture has been fixed ... please download latest version.

Also please check this for streaming issues:

[https://github.com/DroneEngage/droneengage_camera/discussions/3](https://github.com/DroneEngage/droneengage_camera/discussions/3)
